### PR TITLE
[sql] Implement create-table-as-select and add parser/AST support for alter-table

### PR DESCRIPTION
**Pull request by bennostein**
This is in preparation for some larger schema migration changes, but mostly stands on its own, adding a full implementation for create-table-as-select SQL statements and syntactic support for alter-table statements.

### CODE REVIEW

File: sql/src/SqlEval.sk [hunk 1]
```diff
@@ -475,8 +475,63 @@ class Evaluator{options: Options, user: ?UserFile} {
...
       })
-    | P.CreateTableAs _ ->
-      invariant_violation("CREATE TABLE AS SELECT not implemented")
+    | P.CreateTableAs{pos, name, select, ifNotExists} ->
+      core: P.SelectCoreQuery = select.core match {
+      | P.SelectCoreValues _ ->
+        error(
+          pos,
+          "CREATE TABLE AS requires a SELECT query, not a VALUES literal",
+        )
```
Comment (beauby) [inline sql/src/SqlEval.sk:484]: Note that SQLite accepts the following:
```
CREATE TABLE bar AS VALUES(1, 'hello')
```
and gives the anonymous columns names like `column1`, `column 2`.

File: sql/src/SqlEval.sk [hunk 2]
```diff
@@ -475,8 +475,63 @@ class Evaluator{options: Options, user: ?UserFile} {
...
+
+      (_, cols) = SKDB.Compiler::getCols(core.params, cselect.from);
+
+      schema = cselect.params.mapWithIndex((idx, param) -> {
+        P.ColumnDefinition{
+          name => cols.maybeGet(idx) match {
+          | None() ->
+            error(
+              pos,
+              "Anonymous columns not allowed in CREATE TABLE AS SELECT",
```
Comment (beauby) [inline sql/src/SqlEval.sk:504]: SQLite also accepts anonymous expressions here, and simply sets the column name to the string representation of the expression:
```
CREATE TABLE foo AS SELECT 2+2;
# Is equivalent to:
# CREATE TABLE foo ("2+2" INT);
# INSERT INTO foo VALUES(2+2);
```

Comment (bennostein) [reply sql/src/SqlEval.sk:504]: Right, I thought that being more restrictive than sqlite made sense for the schema migration application -- one can imagine mistakenly creating a table with anonymous columns and then being stuck with it.

But I can easily remove those restrictions for general uses and just enforce non-anonymous columns for schema migrations in particular.

Comment (beauby) [reply sql/src/SqlEval.sk:504]: I think it's fine as is, especially since it's failing with a clear error message.

File: sql/src/SqlEval.sk (hunk 1 above)
Comment (gregsexton) [reply sql/src/SqlEval.sk:484]: Bleh! I don't think we should support that. Particularly if we don't support renaming cols afterward.

**Comment by gregsexton** [comment]
LGTM

**Pull request merged**